deployment/upgradeLive

```diff
@@ -2,43 +2,133 @@
-    mv liveDVR-$1 $1
-    rm -rf /opt/kaltura/liveController/v$1.tar.gz
-    cd $1
-    echo "current dir: [`pwd`]"
+if [ $# -eq 2 ]; then
+        PARAM=$2
+fi
+
+function download_new_version() {
+	rm -rf "v$VERSION.tar.gz.1*"
+    echo starting to download  https://github.com/kaltura/liveDVR/archive/v$VERSION.tar.gz
+    wget https://github.com/kaltura/liveDVR/archive/v$VERSION.tar.gz
+    echo running unzip v1.14.6.tar.gz
+    tar -xvzf v$VERSION.tar.gz
+    mv liveDVR-$VERSION $VERSION
+    rm -f /opt/kaltura/liveController/v$VERSION.tar.gz
+}
+
+function replace_upgrade_script() {
+	echo "replacing upgradeLive script with v$VERSION"
+    if [ ! -r v$VERSION.tar.gz ]; then
+        echo "starting to download v$VERSION.tar.gz"
+        wget https://github.com/kaltura/liveDVR/archive/v$VERSION.tar.gz
+    else
+        echo "v$VERSION.tar.gz found, skiping download"
+    fi
+    echo "extracting liveDVR-$VERSION/deployment/upgradeLive from liveDVR-$VERSION.tar.gz"
+    tar -zxvf v$VERSION.tar.gz liveDVR-$VERSION/deployment/upgradeLive
+    echo "copying  liveDVR-$VERSION/deployment/upgradeLive to /etc/init.d/"
+    cp -f liveDVR-$VERSION/deployment/upgradeLive /etc/init.d/
+    echo "removing artifacts..."
+    rm -rf v$VERSION.tar.gz*
+    rm -rf liveDVR-$VERSION
+}
+
+function install() {
+    cd $VERSION
     nvm install
     # move configuration files (liveController's and liveRecorder's) to latest version
-    cp /opt/kaltura/liveController/configMapping.json /opt/kaltura/liveController/$1/common/config/
-    cp /opt/kaltura/liveController/configMapping.ini /opt/kaltura/liveController/$1/liveRecorder/Config/
+    cp /opt/kaltura/liveController/configMapping.json /opt/kaltura/liveController/$VERSION/common/config/
+    cp /opt/kaltura/liveController/configMapping.ini /opt/kaltura/liveController/$VERSION/liveRecorder/Config/
     npm install
     cd ..
     # download binaries from Jenkins
     echo "### starting to download binaries"
-    bash /opt/kaltura/liveController/$1/deployment/get_bins "$1"
+    /opt/kaltura/liveController/$VERSION/deployment/get_bins "$VERSION"
     echo "### finished downloading binaries"
     # create symlinks to latest dir and process script
-	if [ -L "latest" ]; then
-		echo "unlink latest"
-		unlink latest
-	fi
-    ln -s /opt/kaltura/liveController/$1 /opt/kaltura/liveController/latest
-	if [ ! -L /etc/init.d/kLiveController ]; then
-		echo "Creating symlink to kLiveController script at - /etc/init.d/kLiveController"
-		ln -s /opt/kaltura/liveController/latest/serviceWrappers/linux/kLiveController /etc/init.d/kLiveController
-	fi
+    if [ -L "latest" ]; then
+         echo "unlink latest"
+         unlink latest
+    fi
+    ln -s /opt/kaltura/liveController/$VERSION /opt/kaltura/liveController/latest
+
+}
+
+function downgrade() {
+    if [ -L "latest" ]; then
+        echo "unlink latest"
+        unlink latest
+    fi
+    ln -s /opt/kaltura/liveController/$VERSION /opt/kaltura/liveController/latest
```

Comment:
we can't assume that machine has the desired version folder. Maybe a day before someone cleaned all the folder from previous version.
In that case we will have to download again the required version